procrastinate/worker.py

```diff
@@ -83,29 +86,38 @@ async def run(self) -> None:
                 "queues": queues,
             },
         )
+        self.notify_event = None
 
     async def single_worker(self):
-        queues = self.queues
-        display = queues_display(queues)
-
         while not self.stop_requested:
             job = await self.job_store.fetch_job(self.queues)
-            if not job:
-                if not self.wait:
+            if job:
+                await self.process_job(job=job)
+            else:
+                if not self.wait or self.stop_requested:
```

Comment:
Out of curiosity, is this `stop_requested` test required here, or is just an optimization or something? I find it weird that we test `stop_requested` at two different places in the loop.

Reply:
It's an optimization, so that we don't wait if stop was requested during the job fetching coroutine.